src/Persistence/Sql/Query.php

```diff
@@ -336,20 +272,18 @@ protected static function fillWhere(DsqlQuery $query, Model\Scope\AbstractScope
 
     public function getIterator(): iterable
     {
-        try {
-            return $this->select()->execute();
-        } catch (\PDOException $e) {
-            throw (new Exception('Unable to execute iteration query', 0, $e))
-                ->addMoreInfo('query', $this->getDebugQuery())
-                ->addMoreInfo('message', $e->getMessage())
-                ->addMoreInfo('model', $this->model)
-                ->addMoreInfo('scope', $this->model->scope()->toWords());
-        }
+        return $this->tryExecute();
```

Comment:
try sounds like we do not throw at all, really? intended?

Reply:
It is exactly the opposite 😄 
wrapping the exception and adding query info to it
any suggections about naming?

Reply:
Mybe I just include the try wrap in the execute method so it always adds the query debug info

Reply:
`execute` :) it does not try anything, it only nest exception with better info - also, `Unable` is wrong, unable means you was not able to so mething instead of something that was able to exec failed

Reply:
yes, figured out execute only is best and implemented it. Exception message also changed